### ISSUE TITLE
IntelliCode for C# Dev Kit will not show line completions regardless of GitHub Copilot state

### ISSUE THREAD

**Issue by riverar** [MicrosoftDocs/intellicode#471]
**Product and Version** [VS/VSCode]: VSCode
```
Version: 1.80.2 (user setup)
Commit: 2ccd690cbff1569e4a83d7c43d45101f817401dc
Date: 2023-07-27T20:40:28.909Z
Electron: 22.3.14
ElectronBuildId: 22695494
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.25915
```

**OS Version** [macOS/Windows]: Windows
**IntelliCode Extension Version**: IntelliCode for C# Dev Kit v0.1.26
**Target Platform or Language** [e.g. Node.js]: Any

**Steps to Reproduce / Scenario:**

1. Install `GitHub.copilot` extension
2. Single-click the GitHub CoPilot icon in the status area to disable it (it should turn yellow)
3. Install `ms-dotnettools.vscodeintellicode-csharp` extension
4. Observe line completions are not available due to "Copilot being enabled" which is not the case

Output
```
Inline Completions document change subscription registered.
Initializing IntelliCode Inline Completions
[Completions][3]: IntelliCode for C# Dev Kit will not show line completions because Copilot is enabled.
IntelliCode Inline Completions was not activated.
```

Screenshot
![image](https://github.com/MicrosoftDocs/intellicode/assets/475132/fa1b77de-514f-4369-8fae-eae68c5d1296)

**Comment by timheuer** [comment]
In addition to this state issue, there have been several reports the dialog is confusing as it purports to appear there is an issue versus being informative (warning icon makes it look like something is wrong).  We should consider fixing the messaging here to be clear if there is action or just informative. We may want to consider if this is even a necessary dialog.

**Comment by justinas-marozas** [comment]
This issue is misclassified. I have the same issue. IntelliCode disables itself even when Copilot completions are also disabled. I should be able to use IntelliCode when Copilot completions are disabled.
<img width="426" alt="image" src="https://github.com/MicrosoftDocs/intellicode/assets/7419290/93d830f0-1694-497b-8e79-e0d1f9442b14">

**Comment by timheuer** [comment]
> This issue is misclassified. I have the same issue. IntelliCode disables itself even when Copilot completions are also disabled. I should be able to use IntelliCode when Copilot completions are disabled. 

Correct, when Copilot is **disabled** you should. Are you seeing that not being the case?

**Comment by riverar** [comment]
@justinas-marozas You just described the original issue 👍. It's a confusing dialog for sure!

**Comment by justinas-marozas** [comment]
> Correct, when Copilot is **disabled** you should. Are you seeing that not being the case?

Thanks for the prompt reply!

Copilot completions **are** disabled, and I'm getting the toast saying it's enabled.
<img width="556" alt="image" src="https://github.com/MicrosoftDocs/intellicode/assets/7419290/594d786a-e586-4c56-b81a-81e7f66d5cb3">

But the problem seems to be project-dependant. When I first commented here, my solution only had one project (consoleapp) and in that project I'm getting no help from IntelliCode whatsoever, but I now have created a second project (xunit) and IntelliCode seems to work there.

I'm working in WSL: Ubuntu-22.04.

  Version: 1.84.2 (user setup)
  Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
  Date: 2023-11-09T10:51:52.184Z
  Electron: 25.9.2
  ElectronBuildId: 24603566
  Chromium: 114.0.5735.289
  Node.js: 18.15.0
  V8: 11.4.183.29-electron.0
  OS: Windows_NT x64 10.0.22621
  
  IntelliCode for C# Dev Kit v0.1.26

**Comment by timheuer** [comment]
Ok, thanks...seems legit obvious something amiss here. Will ping some folks on this.

**Comment by miqm** [comment]
I've same problem with a C# project with aspnet razor pages. Copilot is useless against razor so I want to have "old-style" completions. Even though I disabled copilot on aspnetrazor pages, I still don't get intellisense suggestions.
![Screenshot 2024-01-09 at 14 20 34](https://github.com/MicrosoftDocs/intellicode/assets/7921224/bf4cdbff-9974-47e9-ab20-b83f713188f0)

**Comment by LiamColeman** [comment]
I am also having this issue. I disabled the entire copilot extension as a whole in the extensions page and my completions are back for now.

**Comment by masoudmodabber** [comment]
I am very dependant on Copilot but at the same I want c# dev kit to do the auto-completion as copilot auto-completion isn't good. That would be rly great if such an option is available and let us choose which tool we want to choose forr auto-completion.

**Comment by riverar** [comment]
@timheuer Anniversary for this issue is near.

**Comment by timheuer** [comment]
> I am very dependant on Copilot but at the same I want c# dev kit to do the auto-completion as copilot auto-completion isn't good. That would be rly great if such an option is available and let us choose which tool we want to choose forr auto-completion.

You have this today depending on what you mean by Auto-completion...

- if you mean typical IntelliSense, this does not block that
- if you mean "big block of completions, multi-line suggestions" -- right now in VSCode, there is one provider...and we've chosen to say "if you have copilot enabled, we prioritize that intent" and suppress the IntelliCode 'big block' completions (which we call whole-line completions)
- "Starred" completions (intellisense with the ⭐ next to them) are still available even with Copilot enabled.

It really is just the bigger block multi-line/whole-line suggestions that one provider needs to handle.

**Comment by ohadcn** [comment]
> if you mean typical IntelliSense, this does not block that

what is not working for me right now, is a type aware completion.

so if I write:

    string a = "Hello World!";
    a.

now copilot tries to offer me something (that's OK), but when I press `ctrl + space` - I want to get list of `string` class properties and methods, while right now I get list with every name in .net builtins or in my project.

![Screenshot from 2024-03-21 14-04-48](https://github.com/MicrosoftDocs/intellicode/assets/905635/fad55202-3eac-4206-a486-b94359311eaf)

**Comment by miqm** [comment]
I had this problem and I fixed it by installing dotnet on mac from installer - not brew - and allowing vscode to detect where dotnet is. Also, I created a solution where I added projects in my Workspace. Then it started to work, although not sure if only one step is required or all.
